sdks/java/core/src/test/java/org/apache/beam/sdk/util/ReshuffleTest.java

```diff
@@ -88,6 +94,33 @@ public void testJustReshuffle() {
     pipeline.run();
   }
 
+  /**
+   * Tests that timestamps can
+   */
+  @Test
+  @Category(RunnableOnService.class)
+  public void testReshuffleAndAssignTimestamps() {
+    PCollection<KV<String, Integer>> input = pipeline
+        .apply(Create.of(ARBITRARY_KVS)
+            .withCoder(KvCoder.of(StringUtf8Coder.of(), VarIntCoder.of())));
+
+    PCollection<KV<String, Integer>> output =
+        input
+            .apply(Reshuffle.<String, Integer>of())
+            .apply(
+                ParDo.of(
+                    new DoFn<KV<String, Integer>, KV<String, Integer>>() {
+                      @ProcessElement
+                      public void assertMinValueTimestamp(ProcessContext ctxt) {
+                        assertThat(ctxt.timestamp(), equalTo(BoundedWindow.TIMESTAMP_MIN_VALUE));
```

Comment:
TIMESTAMP_MIN_VALUE feels like a too special value. If you want to assert that the original timestamps are being preserved, perhaps better to supply test data using Create.timestamped()? And then perhaps apply a ParDo that reifies timestamps (or do we already have a transform for that?) and PAssert.that(result).containsInAnyOrder(original timestamped data)?